DSL/conditions.md

```diff
@@ -0,0 +1,288 @@
+# Declarative DSL to Check Conditions in JSON Schema
+
+This document provides an overview of a declarative Domain Specific Language (DSL) designed to check various conditions in a JSON schema. The DSL is structured as an array of conditions, with each condition containing three mandatory fields: `path`, `operation`, and `value`.
+
+## DSL Structure
+
+### Condition Fields
+
+- **path**: Specifies the path to the target in the schema where the operation will be performed.
+- **operation**: Specifies the type of operation to perform at the target path.
+- **value**: Specifies the value to be used in the operation.
+
+## Supported Operations
+
+### Type Checks
+
+- **type-is**: Checks if the type of the target matches the specified type.
+- **contains-type**: Checks if an array contains elements of a specified type.
+
+### Equality Checks
+
+- **equals**: Checks if the target equals the specified value.
+- **not-equals**: Checks if the target does not equal the specified value.
+- **size-equals**: Check if the size of the array or number of properties of the object equal the specified value. 
+
+### Presence Checks
+
+- **has-key**: Checks if the target object has the specified key.
+
+### Array Operations
+
+- **contains**: Checks if the array contains the specified value.
+
+### Object Operations
+
+- **min-properties**: Checks if the object has a minimum number of properties.
+
+### Numeric Operations
+
+- **maximum**: Checks if the numeric target is less than or equal to the specified value.
+- **minimum**: Checks if the numeric target is greater than or equal to the specified value.
+
+### String Operations
+
+- **match-pattern**: Checks if the string target matches the specified regex pattern.
+
+## Instances of majorly used cases:
```

Comment:
No colons in titles. Also sounds like you should call this "Examples" or something like that

Reply:
Actually, instead of providing these higher level examples here, I think you should just add simpler examples for every operation above. If you explain the individual operations well, I think it might be clear enough how they can be tied together?